### PR TITLE
Run pydocstyle as a test on Travis CI

### PR DESCRIPTION
**Pull request by sgillies**
For now, we're only selecting D1xx.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,10 +6,10 @@ python:
   - "3.5"
   - "3.6"
 install:
-  - "pip install pytest-cov"
-  - "pip install -e ."
+  - "pip install -e .[test]"
```
Comment (sgillies) [inline .travis.yml]: Less Travis config is better, and "test" as name for a set of extra testing requirements is fairly standard, see https://github.com/pypa/sampleproject/blob/master/setup.py.

File: .travis.yml [hunk 2]
```diff
@@ -6,10 +6,10 @@ python:
...
 script:
   - "python -m pytest --cov mercantile --cov-report term-missing"
+  - "python -m pydocstyle mercantile"
```
Comment (sgillies) [inline .travis.yml]: We're now running pydocstyle as a test that must be passed.

File: mercantile/scripts/__init__.py [hunk 3]
```diff
@@ -11,6 +11,17 @@
...
+
+    Parameters
+    ----------
+    verbosity : int
+        The number of `-v` options from the command line.
+
+    Returns
+    -------
+    None
+    """
```
Comment (sgillies) [inline mercantile/scripts/__init__.py]: Pydocstyle found this missing docstring.

File: mercantile/scripts/__init__.py [hunk 4]
```diff
@@ -60,6 +71,7 @@ def iter_lines(lines):
 @click.version_option(version=mercantile.__version__, message='%(version)s')
 @click.pass_context
 def cli(ctx, verbose, quiet):
+    """Execute the main mercantile command"""
```
Comment (sgillies) [inline mercantile/scripts/__init__.py]: And this one.

File: setup.cfg [hunk 5]
```diff
@@ -0,0 +1,2 @@
+[pydocstyle]
+select = D1
```
Comment (sgillies) [inline setup.cfg]: @perrygeo @dnomadb I think that starting at D1xx and increasing incrementally could be a good pattern.

File: setup.py [hunk 6]
```diff
@@ -6,28 +6,35 @@
...
+    extras_require={
+        "dev": ["check-manifest"],
+        "test": ["coveralls", "pytest-cov", "pydocstyle"],
+    },
+    entry_points="""
       [console_scripts]
       mercantile=mercantile.scripts:cli
-      ''')
+      """,
+)
```
Comment (sgillies) [inline setup.py]: I ran black on the setup script.

**Comment by sgillies** [comment]
@dnomadb @perrygeo ready for review.

File: .travis.yml (hunk 2 above)
Comment (perrygeo) [reply .travis.yml]: I'm not sure docstrings are essential for test code. My theory is that the test name itself should be descriptive enough on its own.

File: setup.py [hunk 7]
```diff
@@ -6,28 +6,35 @@
...
-      url='https://github.com/mapbox/mercantile',
-      license='BSD',
-      packages=find_packages(exclude=['ez_setup', 'examples', 'tests']),
-      include_package_data=True,
-      zip_safe=False,
-      install_requires=['click>=3.0'],
-      entry_points='''
+setup(
+    name="mercantile",
+    version="1.0.2",
```
Comment (dnomadb) [inline setup.py]: Should be 1.0.4 correct?

File: setup.py [hunk 8]
```diff
@@ -13,7 +13,7 @@
 
 setup(
     name="mercantile",
-    version="1.0.2",
+    version="1.0.4",
```
Comment (sgillies) [inline setup.py]: Got it @dnomadb .

**Pull request merged**